src/User/User.php

```diff
@@ -128,13 +176,15 @@ public function setIdentity(IdentityInterface $identity): void
      *
      * @param IdentityInterface $identity the user identity (which should already be authenticated)
      * @param int $duration number of seconds that the user can remain in logged-in status, defaults to `0`
+     * @param ResponseInterface $response Response to handle
      * @return bool whether the user is logged in
      */
-    public function login(IdentityInterface $identity, int $duration = 0): bool
+    public function login(IdentityInterface $identity, int $duration = 0, ResponseInterface $response): bool
```

Comment:
Setting response here isn't a good idea overall. I think it is middleware job.

Reply:
If I remove the response out of there, I have no way to add the cookie. I could add the response as a parameter of the middleware process method, that way I get the response and I can pass it to the login so that it saves the cookie. Now if I remove that parameter from the login method, I can't do that.